File: dotnet/GameServerSimulation/Probability/Box.cs
namespace GameServerSimulation.Probability
{
    public class Box
    {
        public int Count { get; internal set; }

        public Reward Reward { get; private set; }

        public Box(Reward reward, int count)
        {
            Reward = reward;
            Count = count;
        }

        public Box(Box other)
        {
            Reward = other.Reward;
            Count = other.Count;
        }

        public override string ToString() => $"{Reward} {Count}";
    }
}

File: java/src/main/java/com/yggdrasilAssignment/probability/Box.java
package com.yggdrasilAssignment.probability;

import com.yggdrasilAssignment.Reward;

public class Box {
    private int _count;
    private final Reward _reward;

    public Box(Reward reward, int count) {
        _reward = reward;
        _count = count;
    }

    public Box(Box other) {
        _reward = other.getReward();
        _count = other.getCount();
    }

    public Reward getReward() {
        return _reward;
    }

    public int getCount() {
        return _count;
    }

    public void decrementCount() {
        _count--;
    }
}
